### PR TITLE
Enable building sourcebuild

### PR DESCRIPTION
**Pull request by 333fred**
This conditions our build checks that are irrelevant when building sourcebuild, and adds a minimal solution for sourcebuild to build. Next steps will be to enable PR checks to ensure that this doesn't break, this is tracked by https://github.com/dotnet/roslyn/issues/26044.

<details><summary>Ask Mode template completed</summary>

### Customer scenario

Integrates infrastructure changes to enable building Sourcebuild releases.

### Bugs this fixes

Roslyn side of https://github.com/dotnet/source-build/issues/356.

### Workarounds, if any

Patching all of this in in the Sourcebuild repo. This is fragile and _will_ break at some point.

### Risk

None, this doesn't affect shipping code.

### Performance impact

None, this doesn't affect shipping code.

### Is this a regression from a previous update?

No.

### Root cause analysis

N/A

### How was the bug found?

N/A

### Test documentation updated?

N/A
</details>

### CODE REVIEW

**Comment by jaredpar** [comment]
Can you add SourceBuild.sln to the set of solutions that BuildBoss monitors? Just add it here 

https://github.com/dotnet/roslyn/blob/master/build/scripts/test-build-correctness.ps1#L42

File: NuGet.Config [hunk 1]
```diff
@@ -12,6 +12,7 @@
   </solution>
   <packageSources>
     <clear />
+    <add key="source-built" value="C:\Users\frsilb\Documents\git\source-build\bin/obj/x64/Release/blob-feed/packages/" />
```
Comment (CyrusNajmabadi) [inline NuGet.Config]: hrmm?

Comment (333fred) [reply NuGet.Config]: Whoops. Added more files than I wanted, I'll amend the commit.

Comment (333fred) [reply NuGet.Config]: Fixed.

**Comment by 333fred** [comment]
Ping @dotnet/roslyn-infrastructure, I need a second review on this please.

**Comment by jaredpar** [comment]
Infra changes only need one review IIRC.

**Comment by 333fred** [comment]
Got it. I'm used to needing 2.

**Comment by 333fred** [comment]
@dotnet-bot retest windows_release_vs-integration_prtest please

**Comment by 333fred** [comment]
Test failure was https://github.com/dotnet/roslyn/issues/26037

**Pull request merged**